core/jvm/src/main/scala/com/dwolla/fs2utils/hashing/Sha256Pipe.scala

```diff
@@ -7,7 +7,7 @@ import cats.effect.concurrent.Deferred
 import fs2._
 
 object Sha256Pipe {
-  def apply[F[_] : Sync](promisedHexString: Deferred[F, String]): Pipe[F, Byte, Byte] = {
+  def apply[F[_] : Sync](promisedHexString: Deferred[F, Either[Throwable, String]]): Pipe[F, Byte, Byte] = {
```

Comment:
`Deferred[F, *]`s do not have an error channel, even if the underlying `F[_]` does, so the output type needs to carry the possibility of error.